sdk/core/src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryPlanManager.java

```diff
@@ -151,4 +148,21 @@ private Block createBlock(Protos.TaskInfo taskInfo)
                 recoveryRequirements.get(0),
                 launchConstrainer);
     }
+
+    @Override
+    public Set<String> getDirtyAssets() {
+        Set<String> dirtyAssets = new HashSet<>();
+        if (plan != null) {
+            final List<? extends Phase> phases = plan.getChildren();
```

Comment:
I know this is rich coming from me, but can't this be done more succinctly in a streaming style?